### PR TITLE
TRUNK-247:Adding "required values" to our java Objects

### PR DESCRIPTION
**Pull request by reagan-meant**
TRUNK-247:Adding "required values" to our java Objects
I added constructors that allow parameters which are not nullable..
see https://issues.openmrs.org/projects/TRUNK/issues/TRUNK-247
## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/BaseConceptMap.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
 	private Date dateCreated;
 	
 	private Date dateChanged;
-	
+
```
Comment (dilanthas) [inline api/src/main/java/org/openmrs/BaseConceptMap.java]: unwanted empty line

File: api/src/main/java/org/openmrs/ConceptDatatype.java [hunk 2]
```diff
@@ -79,8 +79,13 @@ public ConceptDatatype(Integer conceptDatatypeId) {
...
 	}
 	
+	public ConceptDatatype(Integer conceptDatatypeId, String hl7Abbreviation) {
+		super();
+		this.conceptDatatypeId = conceptDatatypeId;
+		this.hl7Abbreviation = hl7Abbreviation;
+	}
 	// Property accessors
-	
+
```
Comment (HerbertYiga) [inline api/src/main/java/org/openmrs/ConceptDatatype.java]: unwanted empty line still here

**Comment by dkayiwa** [comment]
@reagan-meant can you add a link to the ticket as advised here? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by mozzy11** [comment]
@reagan-meant  did you see the merge comflicts ?

**Comment by reagan-meant** [comment]
@mozzy11  Its been a while..I will check it out and see.Thanks

File: api/src/main/java/org/openmrs/ConceptDatatype.java (hunk 2 above)
Comment (ujjwaltyagi355) [reply api/src/main/java/org/openmrs/ConceptDatatype.java:87]: Yup.

File: api/src/main/java/org/openmrs/ConceptSet.java [hunk 3]
```diff
@@ -41,13 +41,25 @@
...
 	
+	public ConceptSet(Integer conceptSetId) {
+		this.conceptSetId = conceptSetId;
+	}
+
 	public ConceptSet(Concept concept, Double weight) {
 		setConcept(concept);
 		setSortWeight(weight);
 	}
-
```
Comment (ujjwaltyagi355) [inline api/src/main/java/org/openmrs/ConceptSet.java:48]: The extra line here could be removed.

File: api/src/main/java/org/openmrs/ConceptSet.java [hunk 4]
```diff
@@ -41,13 +41,25 @@
...
+	public ConceptSet(Integer conceptSetId, Concept concept, Concept conceptSet, User creator, Date dateCreated) {
+		this.conceptSetId = conceptSetId;
+		this.concept = concept;
+		this.conceptSet = conceptSet;
+		this.creator = creator;
+		this.dateCreated = dateCreated;
+	}
 	// Property accessors
 	
+
```
Comment (ujjwaltyagi355) [inline api/src/main/java/org/openmrs/ConceptSet.java:62]: Is this line required?

File: api/src/main/java/org/openmrs/Drug.java [hunk 5]
```diff
@@ -65,8 +65,13 @@ public Drug(Integer drugId) {
 		this.drugId = drugId;
 	}
 	
-	// Property accessors
```
Comment (ujjwaltyagi355) [inline api/src/main/java/org/openmrs/Drug.java:68]: Can remove this extra line.

File: api/src/main/java/org/openmrs/DrugIngredient.java [hunk 6]
```diff
@@ -32,8 +32,16 @@
 	public DrugIngredient() {
 	}
 	
+	public DrugIngredient(Drug drug) {
+		this.drug = drug;
+	}
+
+	public DrugIngredient(Drug drug, Concept ingredient) {
```
Comment (ujjwaltyagi355) [inline api/src/main/java/org/openmrs/DrugIngredient.java:39]: Maybe we should add "strength" and "units" as parameters as well.

File: api/src/main/java/org/openmrs/FieldType.java [hunk 7]
```diff
@@ -32,9 +32,13 @@ public FieldType() {
 	public FieldType(Integer fieldTypeId) {
 		this.fieldTypeId = fieldTypeId;
 	}
-	
+		
+	public FieldType(Integer fieldTypeId, Boolean isSet) {
+		this.fieldTypeId = fieldTypeId;
+		this.isSet = isSet;
+	}
 	// Property accessors
```
Comment (ujjwaltyagi355) [inline api/src/main/java/org/openmrs/FieldType.java:40]: An extra line may be added here.

File: api/src/main/java/org/openmrs/Form.java [hunk 8]
```diff
@@ -52,6 +52,11 @@ public Form(Integer formId) {
 		this.formId = formId;
 	}
 	
+	public Form(Integer formId, String version) {
```
Comment (ujjwaltyagi355) [inline api/src/main/java/org/openmrs/Form.java:55]: Can we take "published" as well here?

File: api/src/main/java/org/openmrs/OrderFrequency.java [hunk 9]
```diff
@@ -28,6 +28,19 @@
 	
 	private Concept concept;
 	
+
```
Comment (ujjwaltyagi355) [inline api/src/main/java/org/openmrs/OrderFrequency.java:31]: Maybe this extra space could be removed.

File: api/src/main/java/org/openmrs/OrderGroup.java [hunk 10]
```diff
@@ -35,6 +35,22 @@
 	
 	private OrderSet orderSet;
 	
+	
+
```
Comment (ujjwaltyagi355) [inline api/src/main/java/org/openmrs/OrderGroup.java:39]: Maybe these extra lines can be removed.

File: api/src/main/java/org/openmrs/OrderSetMember.java [hunk 11]
```diff
@@ -29,6 +29,22 @@
 	
 	private Concept concept;
 	
+	
+
```
Comment (ujjwaltyagi355) [inline api/src/main/java/org/openmrs/OrderSetMember.java:33]: Extra Spaces here.

**Pull request closed without merging**